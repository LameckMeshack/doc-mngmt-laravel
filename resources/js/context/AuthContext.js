// context api to get user from local storage
import React, { createContext, useContext, useEffect, useState } from "react";

export const AuthContext = createContext();

export const useAuth = () => {
    return useContext(AuthContext);
};

export const AuthProvider = ({ children }) => {
    const [user, setUser] = useState(null);

    useEffect(() => {
        const userInfo = localStorage.getItem("userInfo");
        if (userInfo) {
            setUser(JSON.parse(userInfo));
        }
    }, []);

    return (
        <AuthContext.Provider value={{ user, setUser }}>
            {children}
        </AuthContext.Provider>
    );
};
