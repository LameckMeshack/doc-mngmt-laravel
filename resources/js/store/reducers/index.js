import { combineReducers } from "redux";
import authReducer from "./authReducers";
import documentReducer from "./documentReducers";

//combining reducers
export const rootReducer = combineReducers({
    userInfo: authReducer,
    documents: documentReducer,
});
