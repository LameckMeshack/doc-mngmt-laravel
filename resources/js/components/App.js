import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Home from "./Home";
import Login from "./Login";
import Register from "./Register";
import Navbar from "./Navbar";
import { Provider } from "react-redux";
import store from "../store/store";

function App() {
    return (
        <>
            <Navbar />
            <Routes>
                <Route path="/" element={<Home />} />
                <Route path="/login" element={<Login />} />
                <Route path="/register" element={<Register />} />
            </Routes>
        </>
    );
}

export default App;

if (document.getElementById("app")) {
    ReactDOM.render(
        <BrowserRouter>
            <Provider store={store}>
                <App />
            </Provider>
        </BrowserRouter>,

        document.getElementById("app")
    );
}
