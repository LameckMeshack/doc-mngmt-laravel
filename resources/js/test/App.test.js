import React from "react";
import { render } from "@testing-library/react";
import App from "../components/App";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../store/store";
import { AuthProvider } from "../context/AuthContext";

it("should take a snapshot", () => {
    const { asFragment } = render(<App />);

    expect(
        asFragment(
            <Provider store={store}>
                <AuthProvider>
                    <MemoryRouter>
                        <App />
                    </MemoryRouter>
                </AuthProvider>
            </Provider>
        )
    ).toMatchSnapshot();
});
