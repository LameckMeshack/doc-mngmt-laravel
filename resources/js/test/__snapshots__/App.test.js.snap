// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <a
      aria-current="page"
      class="brand-name active"
      href="/"
    >
      <h4>
         DOCUMENT MANAGEMENT
      </h4>
    </a>
    <button
      class="hamburger"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        />
      </svg>
    </button>
    <div
      class="navigation-menu"
    >
      <ul>
        <li>
          <a
            class="a"
            href="/login"
          >
            Login
          </a>
        </li>
        <li>
          <a
            class="a"
            href="/register"
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div>
    <div
      class="document-card-container"
    >
      <h1>
        Files
      </h1>
      <div
        class="cards-container"
      />
    </div>
  </div>
</DocumentFragment>
`;
